contracts/harvester/Harvester.sol

```diff
@@ -12,117 +12,199 @@ import "./../library/BocRoles.sol";
+            );
+            usdStrategies.set(_strategy, sellInfo);
+
+            emit CollectStrategyReward(_strategy,_rewardTokens,_rewardAmounts,_sellTo,_recipient);
         }
     }
 
-    /// @dev Only Keeper can call
-    /// @inheritdoc IHarvester
-    function exchangeAndSend(IExchangeAggregator.ExchangeToken[] calldata _exchangeTokens)
-        external
-        override
-        isKeeperOrVaultOrGovOrDelegate
-    {
-        address _sellToCopy = sellTo;
+    /// @notice Collect the reward token from strategy.
+    /// @param _strategies The target strategies
+    function collectEthStrategies(
+        address[] calldata _strategies
+    ) external override isKeeperOrVaultOrGovOrDelegate {
+        require(ethStrategies.length() == 0, "The sale list has not been processed");
+        for (uint256 i = 0; i < _strategies.length; i++) {
+            address _strategy = _strategies[i];
+            IVault(ethVaultAddress).checkActiveStrategy(_strategy);
+            (
+                address[] memory _rewardTokens,
+                uint256[] memory _rewardAmounts,
+                address _sellTo,
+                bool _needReInvest
+            ) = IStrategy(_strategy).collectReward();
+            address _recipient = _needReInvest ? _strategy : ethVaultAddress;
+            IterableSellInfoMap.SellInfo memory sellInfo = IterableSellInfoMap.SellInfo(
+                _strategy,
+                _rewardTokens,
+                _sellTo,
+                _rewardAmounts,
+                _recipient
+            );
+            ethStrategies.set(_strategy, sellInfo);
+
+            emit CollectStrategyReward(_strategy,_rewardTokens,_rewardAmounts,_sellTo,_recipient);
+        }
+    }
+
+    /// @notice Exchange USD strategy's reward token to sellTo,and send to recipient
+    /// @param _strategy The target strategy
+    /// @param _exchangeTokens The exchange info
+    function exchangeUsdStrategyReward(
+        address _strategy,
+        IExchangeAggregator.ExchangeToken[] calldata _exchangeTokens
+    ) external override isKeeperOrVaultOrGovOrDelegate {
+        IterableSellInfoMap.SellInfo memory sellInfo = usdStrategies.get(_strategy);
+        for (uint256 i = 0; i < _exchangeTokens.length; i++) {
+            IExchangeAggregator.ExchangeToken memory _exchangeToken = _exchangeTokens[i];
+            require(_exchangeToken.toToken == sellInfo.sellToToken, "Rewards can only be sold as sellTo");
+            _exchange(
+                _exchangeToken.fromToken,
+                _exchangeToken.toToken,
+                _exchangeToken.fromAmount,
+                sellInfo.recipient,
+                _exchangeToken.exchangeParam
+            );
+        }
+
+        usdStrategies.remove(_strategy);
+    }
+
+    /// @notice Exchange ETH strategy's reward token to sellTo,and send to recipient
+    /// @param _strategy The target strategy
+    /// @param _exchangeTokens The exchange info
+    function exchangeEthStrategyReward(
```

Comment:
Whether 'exchangeEthStrategyReward' can be merged with 'exchangeUsdStrategyReward' function?